### ISSUE TITLE
Questions about evaluating metric ACC

### ISSUE THREAD

**Issue by yichenwang231** [thuml/Time-Series-Library#219]
Thanks for your work, when I reproduced the classification problem, I found that the results obtained by the validation set and the test set during the training process were inconsistent, and the results of the test set after the training was completed were also inconsistent with the results obtained during the training process.
https://github.com/thuml/Time-Series-Library/blob/843ee18239b895b3018e7240002e85a0433e7f8b/exp/exp_classification.py#L128
https://github.com/thuml/Time-Series-Library/blob/843ee18239b895b3018e7240002e85a0433e7f8b/exp/exp_classification.py#L129
When I evaluate the validation set twice like this, the results will be different
`vali_loss, val_accuracy = self.vali(vali_data, vali_loader, criterion)
            vali_loss1, val_accuracy1 = self.vali(vali_data, vali_loader, criterion)
            print(val_accuracy,val_accuracy1 )`
I don't know if it's due to randomness or other questions, but I'm looking forward to your answers.Thank you.

**Comment by wuhaixu2016** [comment]
I'm sorry for not getting back to you sooner. I have tried the code that you posted and got the same accuracy for val_accuracy and val_accuracy1. I guess that this may come from the randomness of devices. You can try to fix all the random seeds with the following codes

def seed_everything(seed):
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
